delfin/drivers/dell_emc/vmax/alert_handler.py

```diff
@@ -17,129 +17,102 @@
 from oslo_log import log
 
 from delfin import exception
+from delfin.common import constants
 from delfin.drivers.dell_emc.vmax import alert_mapper
 
 LOG = log.getLogger(__name__)
 
 
 class AlertHandler(object):
     """Alert handling functions for vmax driver"""
-    default_category = 'New'
-    default_event_type = 'unknown'
-    default_probable_cause = 'Not available'
-    default_severity = 'unknown'
-    default_me_category = 'storage-subsystem'
-    default_location = 'unknown'
 
-    # Mandatory attributes expected in alert info to proceed with model filling
-    mandatory_alert_attributes = ('storage_id', 'storage_name', 'vendor',
-                                  'model', 'connUnitName', 'emcAsyncEventCode')
+    # Translation of trap severity to alert model severity
+    SEVERITY_MAP = {"emergency": constants.Severity.FATAL,
+                    "alert": constants.Severity.CRITICAL,
+                    "critical": constants.Severity.CRITICAL,
+                    "error": constants.Severity.MAJOR,
+                    "warning": constants.Severity.WARNING,
+                    "notify": constants.Severity.WARNING,
+                    "info": constants.Severity.INFORMATIONAL,
+                    "debug": constants.Severity.INFORMATIONAL,
+                    "mark": constants.Severity.INFORMATIONAL}
+
+    # Translation of trap resource type to alert model resource type
+    RESOURCE_TYPE_MAP = {
+        "hub": constants.ResourceType.NETWORK,
+        "switch": constants.ResourceType.NETWORK,
+        "gateway": constants.ResourceType.NETWORK,
+        "converter": constants.ResourceType.NETWORK,
+        "hba": constants.ResourceType.NETWORK,
```

Comment:
What is the factor for doing these mapping? HBA is Network, why?

Reply:
Updated as per latest model